### ISSUE TITLE
[ExpressionLanguage] Add support for static method calls

### ISSUE THREAD

**Issue by mathielen** [symfony/symfony#13278]
Is there a way to use the ExpressionLanguage for static method calls?

I want to do something like this (within a dependency injection service definition xml):
```xml
<argument type="expression">Infrastructure\Utils.isCli()</argument>
```
But of course the Lexer doesn't allow me to use the "\"
```
SyntaxError in Lexer.php line 90:
Unexpected character "\" around position 14.
```

I searched the docs and the source code but there seems to be no way to use static methods with the ExpressionLanguage, or is there?

Thanks

**Comment by stof** [comment]
The first goal of the ExpressionLanguage is to provide a sandboxed execution environment so allowing to call arbitrary static methods is not something we would include in it IMO.

What you could do is registering your own expression language function in your project doing ``call_user_func('Infrastructure\Utils::isCli')``. note however that registering custom functions for the DI expressions is broken in 2.6.0 and 2.6.1 (and was totally impossible in previous versions as the feature was not implemented), so it would require Symfony 2.6.2+ (which means ``2.6@dev`` currently as 2.6.2 is not yet released)

**Comment by mathielen** [comment]
Aright, I see... :(
Thank you for clearing that up for me.

Regards
